File: Java/Fibonacci/src/Fibonacci.java
public class Fibonacci extends Thread {
    int n;
    int result;
    public Fibonacci(int n) {
        this.n = n;
    }

    public void run() {
        if ((n == 0)||(n == 1 )) result = 1;
        else {
            Fibonacci fl = new Fibonacci(n-1);
            Fibonacci f2 = new Fibonacci(n-2);
            fl.start ();
            f2.start ();
            try {
                fl.join ();
                f2.join ();
            } catch ( InterruptedException e){};
            result = fl.getResult () + f2.getResult ();
        }
    }
    public int getResult (){
        return result ;
    }
    public static void main(String [] args) {
        Fibonacci fl = new Fibonacci(10);
        fl.start ();
        try {
            fl.join ();
        } catch ( InterruptedException e){};
        System.out.println("Answer is " + fl.getResult()); 
    }
}

File: Python/Fibonacci/Fibonacci.py
from threading import Thread
class Fibonacci(Thread):
    def __init__(self, n):
        Thread.__init__(self)
        self.n = n
        self._result = 0

    def run(self):
        if self.n == 1 or self.n == 2:
            self.result = 1
        else:
            f1 = Fibonacci(self.n - 1)
            f2 = Fibonacci(self.n - 2)
            f1.start()
            f2.start()
            f1.join()
            f2.join()
            self.result = f1.result + f2.result

    @property
    def result(self):
        return self._result

    @result.setter
    def result(self, value):
        self._result = value


def main():
    f = Fibonacci(10)
    f.start()
    f.join()
    print(f.result)

if __name__ == '__main__':
    main()

